From e989ef6143b222df6eb79bab626fc972d6513ead Mon Sep 17 00:00:00 2001
From: Pooja Prajod <a0132412@ti.com>
Date: Wed, 4 Mar 2015 16:40:26 +0530
Subject: [PATCH 2/2] gstv4l2src: Optimize delay in capture

This patch reduces the delay in capture by reducing
the S_FMT ioctls for determining whether its interlaced capture.
It assumes capture is not interlaced if driver is vip

Signed-off-by: Pooja Prajod <a0132412@ti.com>
---
 sys/v4l2/gstv4l2object.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index c3ed422..77d75c2 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1706,6 +1706,9 @@ gst_v4l2_object_probe_caps_for_format_and_size (GstV4l2Object * v4l2object,
      * causes expensive and slow USB IO, so don't probe them for interlaced
      */
     interlaced = FALSE;
+  } else if (!strcmp ((char *) v4l2object->vcap.driver, "vip")) {
+    /* Interlaced detection using VIDIOC_TRY/S_FMT */
+    interlaced = FALSE;
   } else {
     /* Interlaced detection using VIDIOC_TRY/S_FMT */
     if (!gst_v4l2_object_get_nearest_size (v4l2object, pixelformat,
-- 
1.7.9.5

